spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -437,6 +448,27 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
 			};
 		}
 
+		private void seekPartitions(Collection<TopicPartition> partitions, boolean idle) {
+			Map<TopicPartition, Long> current = new HashMap<>();
+			for (TopicPartition topicPartition : partitions) {
+				current.put(topicPartition,	ListenerConsumer.this.consumer.position(topicPartition));
+			}
+			ConsumerSeekCallback callback = new ConsumerSeekCallback() {
+
+				@Override
+				public void seek(String topic, int partition, long offset) {
+					ListenerConsumer.this.consumer.seek(new TopicPartition(topic, partition), offset);
```

Comment:
> Otherwise we'd get data first and then seek.

Not sure in your position. We have this code:
```
if (!this.autoCommit) {
						processCommits();
					}
					processSeeks();
					if (this.logger.isTraceEnabled()) {
						this.logger.trace("Polling (paused=" + this.paused + ")...");
					}
					ConsumerRecords<K, V> records = this.consumer.poll(this.containerProperties.getPollTimeout());
```

So, in the main loop we seek first and only after that poll.